### PR TITLE
Introduce integration tests

### PR DESCRIPTION
**Pull request by dennismeister93**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+      - name: Identify changes in devenv repo
+        uses: dorny/paths-filter@v2
+        id: changesInPath
+        with:
+          filters: |
+            setupCommon:
+              - 'setup/src/common/**'
+            setupLang:
+              - 'setup/src/${{ matrix.language }}/**'
```
Comment (MP91) [inline .github/workflows/ci.yml]: please adapt the folders to the current repo (app, sdk)

File: test/cpp/.velocitas.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+            "version": "hash_id"
+        }
+    ],
+    "variables": {
+        "language": "cpp",
+        "repoType": "app",
+        "appManifestPath": "app/AppManifest.json",
+        "githubRepoId": "eclipse-velocitas/vehicle-app-cpp-template",
+        "appExecutionPlatforms": "linux/amd64",
+        "generatedModelPath": "./app/vehicle_model"
```
Comment (MP91) [inline test/cpp/.velocitas.json]: do we really need appManifestPath and generatedModelPath here?

File: test/python/.velocitas.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+            "version": "hash_id"
+        }
+    ],
+    "variables": {
+        "language": "python",
+        "repoType": "app",
+        "appManifestPath": "app/AppManifest.json",
+        "githubRepoId": "eclipse-velocitas/vehicle-app-python-template",
+        "appExecutionPlatforms": "linux/amd64",
+        "generatedModelPath": "./app/vehicle_model"
```
Comment (MP91) [inline test/python/.velocitas.json]: same question as above

**Review by MP91 — CHANGES_REQUESTED** [review]

File: test/cpp/.velocitas.json (hunk 2 above)
Comment (ghost) [reply test/cpp/.velocitas.json]: They are not needed, totally agree

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+          else
+            echo "Changes"
+            echo "changed=1" >> $GITHUB_OUTPUT
+          fi
+        shell: bash
+
+      - name: Fail if there are changes in setup files detected but sync probably failed
+        if: |
+          (steps.changes.outputs.changed == 0 && steps.changesInPath.outputs.setupCommon == 'true') ||
+          (steps.changes.outputs.changed == 0 && steps.changesInPath.outputs.setupLang == 'true')
```
Comment (MP91) [inline .github/workflows/ci.yml]: since you changed the filters, you also have to adapt the checks

**Review by MP91 — CHANGES_REQUESTED** [review]

**Review by MP91 — APPROVED** [review]
LGTM

**Pull request merged**